src/EaShop.Api/Controllers/GoodsController.cs

```diff
@@ -75,6 +75,52 @@ public async Task<IActionResult> GetGoods([FromRoute] int id)
             return Ok(goods);
         }
 
+        // GET: api/Goods/search/name
+        [HttpGet("search/{name}")]
```

Comment:
Change to POST and pass search criteria in body. Implement separate class for search criteria